document/core/appendix/embedding.rst

```diff
@@ -0,0 +1,526 @@
+.. index:: ! embedding, embedder, implementation, host
+.. _embed:
+
+Embedding
+---------
+
+A WebAssembly implementation will typically be *embedded* into a *host* environment.
+An *embedder* implements the connection between such a host environment and the WebAssembly semantics as defined in the main body of this specification.
+An embedder is expected to interact with the semantics in well-defined ways.
+
+This section defines a suitable interface to the WebAssembly semantics in the form of entry points through which an embedder can access it.
+The interface is intended to be complete, in the sense that an embedder does not need to reference other functional parts of the WebAssembly specification directly.
+
+.. note::
+   On the other hand, an embedder does not need to provide the host environment with access to all functionality defined in this interface.
+   For example, an implementation may not support :ref:`parsing <embed-parse-module>` of the :ref:`text format <text>`.
+
+In the description of the embedder interface, syntactic classes from the :ref:`abstract syntax <syntax>` and the :ref:`runtime's abstract machine <syntax-runtime>` are used as names for variables that range over the possible objects from that class.
+Hence, these syntactic classes can also be interpreted as types.
+
+.. _embed-error:
+
+Failure of an interface operation is indicated by an auxiliary syntactic class:
+
+.. math::
+   \begin{array}{llll}
+   \production{(error)} & \error &::=& \ERROR \\
+   \end{array}
+
+In addition to the error conditions specified explicitly in this section, implementations may also return errors when specific :ref:`implementation limitations <impl>` are reached.
+
+
+.. note::
+   Errors are abstract and unspecific with this definition.
+   Implementations can refine it to carry suitable classifications and diagnostic messages.
+
+
+
+.. index:: allocation, store
+.. _embed-store:
+
+Store
+~~~~~
+
+.. _embed-init-store:
+
+:math:`\F{init\_store}() : \store`
+..................................
+
+* Return the empty :ref:`store <syntax-store>`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{init\_store}() &=& \{ \SFUNCS~\epsilon,~ \SMEMS~\epsilon,~ \STABLES~\epsilon,~ \SGLOBALS~\epsilon \} \\
+   \end{array}
+
+
+
+.. index:: module
+.. _embed-module:
+
+Modules
+~~~~~~~
+
+.. index:: binary format
+.. _embed-decode-module:
+
+:math:`\F{decode\_module}(\byte^\ast) : \module ~|~ \error`
+...........................................................
+
+* If there exists a derivation for the :ref:`byte <syntax-byte>` sequence :math:`\byte^\ast` as a :math:`\Bmodule` according to the :ref:`binary grammar for modules <binary-module>`, yielding a :ref:`module <syntax-module>` :math:`m`, then return :math:`m`.
+
+* Else, return :math:`\ERROR`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{decode\_module}(b^\ast) &=& m && (\iff \Bmodule \longrightarrow m{:}b^\ast) \\
```

Comment:
What does the longrightarrow mean? I can guess, but I don't see where it's defined.

Reply:
Yeah I think this is fine, I'm just not accustomed to reading stuff like this so I usually like to make sure I actually know what the symbols mean (which is why [Index of Semantic Rules](https://webassembly.github.io/spec/appendix/index-rules.html) is so useful). You're right, it's probably OK without further explanation since the text is there.